mxtools/flyer.py

```diff
@@ -167,98 +167,147 @@ def unstage(self):
         self.detector.unstage()
         self.detector.cam.acquire.put(0)
 
+    def update_parameters(self, *args, **kwargs):
+        self.detector_arm(**kwargs)
+        self.configure_detector(**kwargs)
+        self.configure_vector(**kwargs)
+        self.configure_zebra(**kwargs)
+
+    def configure_detector(self, **kwargs):
+        file_prefix = kwargs["file_prefix"]
+        data_directory_name = kwargs["data_directory_name"]
+        self.detector.file.external_name.put(file_prefix)
+        self.detector.file.write_path_template = data_directory_name
+
+    def configure_vector(self, *args, **kwargs):
+        angle_start = kwargs["angle_start"]
+        scanWidth = kwargs["scanWidth"]
+        imgWidth = kwargs["imgWidth"]
+        exposurePeriodPerImage = kwargs["exposurePeriodPerImage"]
+        # scan encoder 0=x, 1=y,2=z,3=omega
+
+        self.vector.sync.put(1)
+        self.vector.expose.put(1)
+
+        if imgWidth == 0:
+            angle_end = angle_start
+            numImages = scanWidth
+        else:
+            angle_end = angle_start + scanWidth
+            numImages = int(round(scanWidth / imgWidth))
+        total_exposure_time = exposurePeriodPerImage * numImages
+        if total_exposure_time < 1.0:
+            self.vector.buffer_time.put(1000)
+        else:
+            self.vector.buffer_time.put(3)
+            pass
+        self.setup_vector_program(
+            num_images=numImages,
+            angle_start=angle_start,
+            angle_end=angle_end,
+            exposure_period_per_image=exposurePeriodPerImage,
+        )
+
+    def configure_zebra(self, *args, **kwargs):
+        angle_start = kwargs["angle_start"]
+        exposurePeriodPerImage = kwargs["exposurePeriodPerImage"]
+        detector_dead_time = kwargs["detector_dead_time"]
+        scanWidth = kwargs["scanWidth"]
+        imgWidth = kwargs["imgWidth"]
+        numImages = kwargs["numImages"]
+        self.zebra_daq_prep()
+        ttime.sleep(1.0)
 
-def configure_flyer(
```

Comment:
What is the new/old workflow when using these? You would have had to call actual_scan() from somewhere, presumably? Now you have to call update_parameters from somewhere?

Reply:
as shown in the pseudocode above, we have replaced all of the code that is used by the arming function and actual_scan and moved it into flyer, so you just update_parameters(), then fly